2023/day1/part2.rs

```diff
@@ -0,0 +1,135 @@
+use regex::Regex;
+use core::panic;
+use std::{borrow::BorrowMut, fs};
+
+fn add_char_to_vec(c: &char, vec: &mut Vec<char>) {
+    vec.push(*c);
+}
+
+fn append_first_and_last_char_to_one_digit(vec: &Vec<char>) -> i32 {
+    let mut my_string: String = String::new();
+    my_string.push(*vec.first().unwrap());
+    my_string.push(*vec.last().unwrap());
+    return my_string.parse::<i32>().unwrap();
+}
+
+fn sum_vec(result: &Vec<i32>) -> i32 {
+    let mut sum = 0;
+    for i in result {
+        sum += i;
+    }
+    return sum;
+}
+
+fn match_string_to_single_digit(to_parse: &str) -> Option<char> {
+    match to_parse {
+        "one"   => return Some('1'),
+        "two"   => return Some('2'),
+        "three" => return Some('3'),
+        "four"  => return Some('4'),
+        "five"  => return Some('5'),
+        "six"   => return Some('6'),
+        "seven" => return Some('7'),
+        "eight" => return Some('8'),
+        "nine"  => return Some('9'),
+        _ => None,
+    }
+} 
+
+fn split_string_to_digits(to_parse: &str) -> Option<Vec<char>> {
+    let mut char_vec: Vec<char> = Vec::new();
+    let regexp = Regex::new(r"(\d|one|two|three|four|five|six|seven|eight|nine)").unwrap();
+
+    // using find_iter() instead of captures() because captures is mainly used for regex with groups
+    for capture in regexp.find_iter(to_parse) {
```

Comment:
Bug in find_iter(), same in capture_iter(): 

```log
[part2.rs:43] "to_parse: {}" = "to_parse: {}"
[part2.rs:43] &to_parse = "eightwothree"
[part2.rs:46] "capture: {}" = "capture: {}"
[part2.rs:46] &capture = Match {
    start: 0,
    end: 5,
    string: "eight",
}
[part2.rs:46] "capture: {}" = "capture: {}"
[part2.rs:46] &capture = Match {
    start: 7,
    end: 12,
    string: "three",
}
[part2.rs:109] &nums_in_line = [
    '8',
    '3',
]
```

Probably internally how the Regex handles matching is really simple meaning it probably splits the string on a match? 

Seems to be the case when looking at https://stackoverflow.com/a/57497310 

Eh not sure how to deal with this currently but might be good to check possibility of alternative ways to deal with **look-around** during regex matches. 

```log
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
regex parse error:
    (?=(\d|one|two|three|four|five|six|seven|eight|nine))
    ^^^
error: look-around, including look-ahead and look-behind, is not supported
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~```

I might want to make one simple one myself with the builder https://docs.rs/regex/latest/regex/struct.RegexBuilder.html however that might be a bit overkill.